subsys/nrf_security/src/drivers/cracen/cracenpsa/src/common.c

```diff
@@ -761,39 +816,28 @@ psa_status_t cracen_load_keyref(const psa_key_attributes_t *attributes, const ui
 	if (PSA_KEY_LIFETIME_GET_LOCATION(psa_get_key_lifetime(attributes)) ==
 	    PSA_KEY_LOCATION_CRACEN) {
 
-		k->prepare_key = cracen_prepare_ik_key;
-		k->clean_key = cracen_clean_ik_key;
+		if(cracen_is_ikg_key(attributes)) {
+			return cracen_load_ikg_keyref(attributes, key_buffer, key_buffer_size, k);
+		}
+
 		k->owner_id = MBEDTLS_SVC_KEY_ID_GET_OWNER_ID(psa_get_key_id(attributes));
 		k->user_data = (uint8_t *)&k->owner_id;
```

Comment:
This could be done before the call to `cracen_load_ikg_keyref()` and removed from that function.

Reply:
I actually considered that but I liked having the key reference for the IKG keys loaded all in one place. I find it a bit easier to understand what is happening with the keys that way.